node/lib/util/merge_util.js

```diff
@@ -370,6 +371,7 @@ const mergeStepPrepare = co.wrap(function *(context) {
         return MergeStepResult.error(errorMessage);                  // RETURN
     }
 
+    // TODO: remove this
```

Comment:
Do we want to remove this now, or?

Reply:
I would rather have users hit errors than have PRS hit errors.

Reply:
actually that is a good idea. Updated: 

Originally `changeSubmodules` (non-bare) does two things, (1) delete/open submdoules according to differences bettween urls read from merged .gitmodules vs urls read from HEAD.  (2) writes the urls into .gitmodules on HEAD.

Now I move the second step outside of `changeSubmodules`, and wirte the "smart merged" urls into .gitmodules just like merge-bare would do.

Reply:
ok changeSubmodules is used by a number of places, so needs more work

Reply:
now, passing the urls to changeSubmodules